channeld/channeld.c

```diff
@@ -2381,38 +2393,57 @@ static void peer_reconnect(struct peer *peer,
 	}
 }
 
-/* Funding has locked in, and reached depth. */
-static void handle_funding_locked(struct peer *peer, const u8 *msg)
+/* ignores the funding_depth unless depth >= minimum_depth
+ * (except to update billboard, and set peer->confirmation_needed). */
+static void handle_funding_depth(struct peer *peer, const u8 *msg)
 {
 	unsigned int depth;
+	struct short_channel_id *scid;
 
-	if (!fromwire_channel_funding_locked(msg,
-					     &peer->short_channel_ids[LOCAL],
-					     &depth))
-		master_badmsg(WIRE_CHANNEL_FUNDING_LOCKED, msg);
+	if (!fromwire_channel_funding_depth(tmpctx,
+					    msg,
+					    &scid,
+					    &depth))
+		master_badmsg(WIRE_CHANNEL_FUNDING_DEPTH, msg);
 
 	/* Too late, we're shutting down! */
 	if (peer->shutdown_sent[LOCAL])
 		return;
+	if (depth < peer->channel->minimum_depth) {
+		if (peer->funding_locked[LOCAL] || peer->confirmation_needed == 0)
+			status_failed(STATUS_FAIL_INTERNAL_ERROR,
+					"funding shouldn't be locked");
+		peer->confirmation_needed = peer->channel->minimum_depth - depth;
 
-	if (!peer->funding_locked[LOCAL]) {
-		status_trace("funding_locked: sending commit index %"PRIu64": %s",
-			     peer->next_index[LOCAL],
-			     type_to_string(tmpctx, struct pubkey,
-					    &peer->next_local_per_commit));
-		msg = towire_funding_locked(NULL,
-					    &peer->channel_id,
-					    &peer->next_local_per_commit);
-		sync_crypto_write(&peer->cs, PEER_FD, take(msg));
-		peer->funding_locked[LOCAL] = true;
-	}
+		billboard_update(peer);
+	} else {
+		if (!scid)
+			status_failed(STATUS_FAIL_INTERNAL_ERROR,
+						"funding channel id should be drived");
+		peer->short_channel_ids[LOCAL] = *scid;
+		if (!peer->funding_locked[LOCAL]) {
+			if (peer->confirmation_needed == 0)
+				status_failed(STATUS_FAIL_INTERNAL_ERROR,
+							"funding should be locked");
+			status_trace("funding_locked: sending commit index %"PRIu64": %s",
+				    	peer->next_index[LOCAL],
```

Comment:
There are spaces in these two lines. They usually show up when doing a `git show`as red blocks and probably caused by your editor doing some kind auto indent.

Reply:
Thank you! I will adjust this and be careful.